core/src/martian/swagger.cljc

```diff
@@ -37,39 +35,40 @@
-       :headers-schema (headers-schema ref-lookup parameters)
-       :response-schemas (response-schemas ref-lookup (:responses swagger-definition))
-       :produces (some :produces [swagger-definition swagger-map])
-       :consumes (some :consumes [swagger-definition swagger-map])
-       :summary (:summary swagger-definition)
-       :swagger-definition swagger-definition
-       ;; todo path constraints - required?
-       ;; :path-constraints {:id "(\\d+)"},
-       ;; {:in "path", :name "id", :description "", :required true, :type "string", :format "uuid"
-       :route-name route-name})))
-
-(defn swagger->handlers [swagger-json]
-  (let [swagger-spec (keywordize-keys swagger-json)]
-    (for [[url-pattern swagger-handlers] (:paths swagger-spec)
-          [method swagger-definition] swagger-handlers
-          :let [handler (->handler swagger-spec (:parameters swagger-handlers) url-pattern method swagger-definition)]
-          :when (some? handler)]
-      handler)))
+(defn swagger->handlers
+  ([swagger-json]
+   (swagger->handlers swagger-json nil))
+  ([swagger-json route-name-sources]
+   (let [swagger-spec (keywordize-keys swagger-json)
+         route-names (atom #{})]
+     (for [[url-pattern swagger-handlers] (:paths swagger-spec)
+           :let [common-parameters (:parameters swagger-handlers)]
+           [method definition] (dissoc swagger-handlers :parameters)
+           :let [route-name (produce-route-name route-name-sources url-pattern method definition)]
+           ;; NB: We only care about routes that have a unique name.
+           :when (and (some? route-name)
+                      (unique-route-name? route-name route-names))
+           :let [path-parts (tokenise-path url-pattern)
+                 ref-lookup (select-keys swagger-spec [:definitions :parameters])
+                 parameters (concat common-parameters (:parameters definition))]]
+       {:path               (str/join path-parts)
+        :path-parts         path-parts
+        ;; TODO: Also parse all path constraints — required?
+        ;; :path-constraints {:id "(\\d+)"},
+        ;; {:in "path", :name "id", :description "", :required true, :type "string", :format "uuid" ...}
+        :method             method
+        :path-schema        (path-schema ref-lookup parameters)
+        :query-schema       (query-schema ref-lookup parameters)
+        :body-schema        (body-schema ref-lookup parameters)
+        :form-schema        (form-schema ref-lookup parameters)
+        :headers-schema     (headers-schema ref-lookup parameters)
+        :response-schemas   (response-schemas ref-lookup (:responses definition))
+        :produces           (some :produces [definition swagger-spec])
+        :consumes           (some :consumes [definition swagger-spec])
+        :summary            (:summary definition)
+        :swagger-definition definition
+        :route-name         route-name}))))
```

Comment:
Redo akin to the `openapi->handlers` fn, which keeps the LoB high (and still produces handlers lazily).